### PR TITLE
MMP-110 Make test setup more resilient, add classes to setup environment

### PR DESCRIPTION
**Pull request by WilliamPinho**
…ilient setup and add loggers

### CODE REVIEW

File: src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+import java.nio.file.Paths;
+
+import static org.junit.Assert.fail;
+
+public class AgentSetup {
+
+  private Logger log;
+  private static final long TEN_SECONDS = 10000;
+  private static final String MULE_HOME_FOLDER_PREFIX = "/mule-enterprise-standalone-";
+  private static final String MULE_VERSION = "3.8.1";
```
Comment (damiansima) [inline src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java]: shouldn't this be a parameter?
At some point in time we may want to test against different mule versions

File: src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    stopMule();
+    killMuleProcesses();
+  }
+
+  private void checkAgentIsAcceptingDeployments() throws InterruptedException, IOException {
+    int tries = 0;
+    boolean acceptingDeployments;
+    do {
+      TemporaryFolder folder = new TemporaryFolder();
+      File dummyFile = folder.newFile("dummy.zip");
```
Comment (damiansima) [inline src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java]: Temp folder and file do not need to be in the loop do they?

File: src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+      AgentClient agentClient = new AgentClient(new SystemStreamLog(), "http://localhost:9999/");
```
Comment (damiansima) [inline src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java]: didn't you make the port a parameter, that could change?

File: src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+        agentClient.deployApplication("dummy", dummyFile);
+        acceptingDeployments = true;
+      } catch (Exception e) {
+        log.info("Agent is not accepting deployments yet. Trying again...");
+        log.error("Cause: " + e.getLocalizedMessage());
+        Thread.sleep(TEN_SECONDS);
+        acceptingDeployments = false;
+        tries++;
+      }
+      if (tries == 30) { // Trying for approximately 30 X 10 s = 300 s = 5 minutes
```
Comment (damiansima) [inline src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java]: make the number of attempts a static constant

File: src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+  }
+
+  public String getAnchorFilePath() {
+    return muleHome + ANCHOR_FILE_RELATIVE_PATH;
+  }
+
+
+
+  private void killMuleProcesses() throws IOException {
+    commands = new String[16];
```
Comment (damiansima) [inline src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java]: you can do List<String> l= new ArrayList();
l.add()
l.toArray

File: src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java (hunk 1 above)
Comment (WilliamPinho) [inline src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java]: Done. This is a parameter received by the AgentSetup constructor.

File: src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java (hunk 2 above)
Comment (WilliamPinho) [inline src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java]: No, neither the agent I guess!

File: src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java (hunk 3 above)
Comment (WilliamPinho) [inline src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java]: Not really, now we hard coded it as 9999 as the default mule-standalone comes set

File: src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java (hunk 4 above)
Comment (WilliamPinho) [inline src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java]: Yep

File: src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java (hunk 5 above)
Comment (WilliamPinho) [inline src/test/java/integrationTests/mojo/environmentSetup/AgentSetup.java]: Makes sense

**Pull request merged**